processors/base/helpers.py

```diff
@@ -58,10 +58,24 @@ def default(self, obj):
 
 
 def clean_dict(raw_dict):
-    """Remove False values from dict.
+    """Remove falsy values from dict.
     """
     cleaned_dict = {}
     for key, value in raw_dict.items():
         if value:
             cleaned_dict[key] = value
     return cleaned_dict
+
+
+def get_optimal_title(*strings):
+    """Return first striped string with enough length or last striped string.
+    """
+    MIN_LENGTH = 10
+    for string in strings:
+        if not string:
+            continue
+        string = string.strip()
+        string = string.strip('.')
```

Comment:
Depending on what's the use case here, `str.strip(string.punctuation + string.whitespace)` might be even better. This would remove any whitespace plus `'!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'`